helm-charts/azure-api-management-gateway/templates/deployment.yaml

```diff
@@ -9,6 +9,8 @@ spec:
   selector:
     matchLabels:
       {{- include "azure-api-management-gateway.selectorLabels" . | nindent 6 }}
+  strategy:
+    {{- toYaml .Values.gateway.deployment.strategy | nindent 4 }}
```

Comment:
Now that I think of it, we should make sure that it only gets specified if the user specifies it. That way, it defaults to Kubernetes defaults and not what we define.

Reply:
This would be similar to this:
```yaml
      {{- with .Values.imagePullSecrets }}
      imagePullSecrets:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

Reply:
```suggestion
  {{- with .Values.gateway.deployment.strategy }}
  strategy:
    {{- toYaml .Values.gateway.deployment.strategy | nindent 4 }}
  {{- end }}
```